samplePDFDUNE/samplePDFDUNEBeamFD.cpp

```diff
@@ -0,0 +1,839 @@
+      em_escale_invsqrt_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(em_escale_invsqrt_fd_pos);
+    }
+    else if (name == "HadResFD") {
+      had_res_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(had_res_fd_pos);
+    }
+    else if (name == "MuResFD") {
+      mu_res_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(mu_res_fd_pos);
+    }
+    else if (name == "NResFD") {
+      n_res_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(n_res_fd_pos);
+    }
+    else if (name == "EMResFD") {
+      em_res_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(em_res_fd_pos);
+    }
+    else if (name == "CVNNumuFD") {
+      cvn_numu_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(cvn_numu_fd_pos);
+    }
+    else if (name == "CVNNueFD") {
+      cvn_nue_fd_pos = *it;
+      FDDetectorSystPointers[func_it] = XsecCov->retPointer(cvn_nue_fd_pos);
+    }
+    
+    else { 
+      std::cerr << "Found a functional parameter which wasn't specified in the xml | samplePDFDUNEBeamFD:" << name << std::endl;
+      throw;
+    }
+  }
+  */
+  
+  MACH3LOG_INFO("-------------------------------------------------------------------");
+}
+
+void samplePDFDUNEBeamFD::SetupSplines() {
+
+  ///@todo move all of the spline setup into core
+  if(XsecCov->GetNumParamsFromDetID(SampleDetID, kSpline) > 0){
+    MACH3LOG_INFO("Found {} splines for this sample so I will create a spline object", XsecCov->GetNumParamsFromDetID(SampleDetID, kSpline));
+    splinesDUNE* DUNESplines = new splinesDUNE(XsecCov);
+    splineFile = (splineFDBase*)DUNESplines;
+    InitialiseSplineObject();
+  }
+  else{
+    MACH3LOG_INFO("Found {} splines for this sample so I will not load or evaluate splines", XsecCov->GetNumParamsFromDetID(SampleDetID, kSpline));
+    splineFile = nullptr;
+  }
+  
+  return;
+}
+
+void samplePDFDUNEBeamFD::SetupWeightPointers() {
+  for (int i = 0; i < (int)dunemcSamples.size(); ++i) {
+    for (int j = 0; j < dunemcSamples[i].nEvents; ++j) {
+      MCSamples[i].ntotal_weight_pointers[j] = 6;
+      MCSamples[i].total_weight_pointers[j] = new const double*[MCSamples[i].ntotal_weight_pointers[j]];
+      MCSamples[i].total_weight_pointers[j][0] = &(dunemcSamples[i].pot_s);
+      MCSamples[i].total_weight_pointers[j][1] = &(dunemcSamples[i].norm_s);
+      MCSamples[i].total_weight_pointers[j][2] = MCSamples[i].osc_w_pointer[j];
+      MCSamples[i].total_weight_pointers[j][3] = &(dunemcSamples[i].rw_berpaacvwgt[j]);
+      MCSamples[i].total_weight_pointers[j][4] = &(dunemcSamples[i].flux_w[j]);
+      MCSamples[i].total_weight_pointers[j][5] = &(MCSamples[i].xsec_w[j]);
+    }
+  }
+}
+
+
+int samplePDFDUNEBeamFD::setupExperimentMC(int iSample) {
+  const char *sampleFile = (mtuple_files[iSample]).c_str();
+  dunemc_base *duneobj = &(dunemcSamples[iSample]);
```

Comment:
its fairly minor, but this is just unnecessary. We really can do without any bare pointers in most of this code:

```c++
auto &duneobj = dunemcSamples[iSample];

//...
duneobj.bla[i] = _bla;
```